pkg/metrics/wal/wal_test.go

```diff
@@ -372,6 +372,24 @@ func TestStorage_TruncateAfterClose(t *testing.T) {
 	require.Error(t, ErrWALClosed, s.Truncate(0))
 }
 
+func BenchmarkAppendExemplar(b *testing.B) {
+	walDir, _ := ioutil.TempDir(os.TempDir(), "wal")
+	defer os.RemoveAll(walDir)
+
+	s, _ := NewStorage(log.NewNopLogger(), nil, walDir)
+	app := s.Appender(context.Background())
```

Comment:
You might want to consider closing the storage here on exit even if it's not strictly necessary